Netskope/netskope-transaction/ingest/parser.yml

```diff
@@ -0,0 +1,243 @@
+          - x-policy-dst-host
+          - x-policy-dst-host-source
+          - x-policy-justification-type
+          - x-policy-justification-reason
+          - x-sc-notification-name
+        delimiter: " "
+
+  - name: set_fields
+
+stages:
+  set_fields:
+    actions:
+      - set:
+          event.kind: "event"
+          event.category: ["network"]
+          event.type: ["info"]
+
+          "@timestamp": "{{ parse_event.message.get('x-cs-timestamp') | to_iso8601 }}"
+
+      - set:
+          user.email: "{{ parse_event.message.get('cs-username')}}"
+        filter: '{{ parse_event.message.get("cs-username") | re_match("^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$") }}'
+
+      - set:
+          user.email: "{{ parse_event.message.get('x-cs-app-from-user')}}"
+        filter: '{{ parse_event.message.get("x-cs-app-from-user") | re_match("^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$") }}'
+
+      - set:
+          user.application: "{{ parse_event.message.get('cs-username') }}"
+
+      - set:
+          network.application: "{{ parse_event.message.get('x-cs-app') }}"
+
+      - set:
+          destination.bytes: "{{ parse_event.message.get('cs-bytes') }}"
+          source.bytes: "{{ parse_event.message.get('sc-bytes') }}"
+          network.bytes: "{{ parse_event.message.get('bytes') }}"
+
+      - set:
+          source.nat.ip: "{{ parse_event.message.get('c-ip') }}"
+        filter: "{{parse_event.message.get('c-ip') | is_ipaddress}}"
+
+      - set:
+          source.ip: "{{ parse_event.message.get('x-cs-userip') }}"
+        filter: "{{parse_event.message.get('x-cs-userip') | is_ipaddress }}"
+
+      - set:
+          destination.ip: "{{ parse_event.message.get('s-ip') }}"
+        filter: "{{parse_event.message.get('s-ip') | is_ipaddress }}"
+
+      - set:
+          file.mime_type: "{{ parse_event.message.get('x-rs-file-type') }}"
+          file.hash.md5: "{{ parse_event.message.get('x-rs-file-md5') }}"
+
+      - set:
+          destination.geo.country_name: "{{ parse_event.message.get('x-s-country') }}"
+          destination.geo.city_name: "{{ parse_event.message.get('x-s-location') }}"
+          destination.geo.region_name: "{{ parse_event.message.get('x-s-region') }}"
+          destination.geo.postal_code: "{{ parse_event.message.get('x-s-zipcode') }}"
+
+      - set:
+          source.geo.country_name: "{{ parse_event.message.get('x-c-country') }}"
+          source.geo.city_name: "{{ parse_event.message.get('x-c-location') }}"
+          source.geo.region_name: "{{ parse_event.message.get('x-c-region') }}"
+          source.geo.postal_code: "{{ parse_event.message.get('x-c-zipcode') }}"
+
+      - set:
+          http.request.method: "{{ parse_event.message.get('cs-method') }}"
+          http.request.mime_type: "{{ parse_event.message.get('cs-content-type') }}"
+
+      - set:
+          http.response.status_code: "{{ parse_event.message.get('sc-status') }}"
+        filter: "{{parse_event.message.get('sc-status') != '-'}}"
```

Comment:
Will `ignored_values` not handle this case?

Reply:
It's crashing, because `http.response.status_code` expects a positive integer. My guess is that `ignored_values` applied after the parser.